Game/Scripts/ComboManager.cpp

```diff
@@ -122,14 +131,21 @@ AttackType ComboManager::CheckAttackInput(bool jumping)
 
 void ComboManager::SuccessfulAttack(float specialCount, AttackType type)
 {
-	if(specialCount < 0 || !specialActivated)
+	if (specialCount < 0 || !specialActivated)
 	{
-		this->specialCount = std::max(0.0f, std::min(this->specialCount + specialCount, maxSpecialCount));
+		this->specialCount = static_cast<int>(
+			std::max(0.0f, std::min(static_cast<float>(this->specialCount) + specialCount, maxSpecialCount)));
```

Comment:
For this, you could use [`std::clamp`](https://en.cppreference.com/w/cpp/algorithm/clamp). It basically does the same, but is less verbose. We unfortunately still need a cast, though.
```suggestion
		this->specialCount = std::clamp(this->specialCount + specialCount, 0, static_cast<int>(maxSpecialCount));
```

Reply:
We need another additional cast because `specialCount` (the parameter of the funcion) is a float, so we need to cast that back to an int to match the three other ints (`std::clamp` does not work with `float`, `int`, `int`, it raises an error such as with std::max).